sympy/plotting/plot.py

```diff
@@ -1555,8 +1555,28 @@ def plot(*args, **kwargs):
     show = kwargs.pop('show', True)
     series = []
     plot_expr = check_arguments(args, 1, 1)
-    series = [LineOver1DRangeSeries(*arg, **kwargs) for arg in plot_expr]
 
+    new_plot = []
+    for expr, limit in plot_expr:
+        _, x1, x2 = limit
+        root_deno = []
+        for den in denoms(expr, x):
+            root_deno.extend(real_roots(den))
```

Comment:
`real_roots` is only going to work with polynomial `den`; something like `sqrt(x - 2)` or `sin(x)` will not work. Perhaps `singularities` can be made more robust for these cases.

Reply:
Instead of `real_roots` can we opt for `solve` to calculate the roots, and considering only real root from the obtained list.
As after changing `real_roots` by `solve` locally I tried plotting `1/(sqrt(x - 2) - 1)` and got the correct output.

Reply:
`solve` method generates root in terms of `pi` also. But while passing `pi` as a limit in the function it is not generating proper output. Is there is any way to do it?

Reply:
evaluate it (e.g. `pi.n()` will give a float that starts as  "3.141592..."; if you want fewer digits you can use `pi.n(3)` to get 3 sig figs or `pi.round(3)` to round to the 1/1000 place (approx).

Reply:
`real_roots` is great if the expression is polynomial in `x`; if not then maybe you could fall back to `solve`.

Reply:
You would have to watch out for periodic functions in denominators since more than 1 period might appear in the range of interest and `solve` doesn't give you all values in a given range. But the `periodicity` function can supply that information. `solveset`, too, might be the way to go: e.g if plotting x/(sin(2*x) - 1/2) in the range of [3,12]
```python
>>> [i.n(3) for i in solveset(sin(2*x)-S.Half,domain=Interval(3,12))]
[3.40, 4.45, 7.59, 6.54, 9.69, 10.7]
```

Reply:
/!\ solutions are not given if that S.Half is replaced with 0.5, however.

Reply:
> /!\ solutions are not given if that S.Half is replaced with 0.5, however.

Can we use `nsimplify` to convert float in expression into rational form and then passing it in `solveset`.
```
>>> [i.n(3) for i in solveset(nsimplify(sin(2*x)-S.Half),domain=Interval(3,12))]
[3.40, 4.45, 7.59, 6.54, 9.69, 10.7]
>>> [i.n(3) for i in solveset(nsimplify(sin(2*x)-1/2),domain=Interval(3,12))]
[3.40, 4.45, 7.59, 6.54, 9.69, 10.7]
>>> [i.n(3) for i in solveset(nsimplify(sin(2*x)-0.5),domain=Interval(3,12))]
[3.40, 4.45, 7.59, 6.54, 9.69, 10.7]